src/main/java/io/boomerang/service/profile/LaunchpadNavigationServiceImpl.java

```diff
@@ -122,6 +126,19 @@ private NavigationResponse getExternalNavigationResponse(String email) {
     HttpEntity<String> requestUpdate = new HttpEntity<>("", headers);
     ResponseEntity<NavigationResponse> response =
         restTemplate.exchange(uriComponents.toUriString(), HttpMethod.GET, requestUpdate, NavigationResponse.class);
+    NavigationResponse result = response.getBody();
+    if (result != null && result.getPlatform() != null) {
+      if(Strings.isBlank(result.getPlatform().getAppName())) {
+        // set default appName if the external Navigation API does NOT return appName.
+        String appName = settingsService.getConfiguration("customizations", "appName").getValue();
+        appName = Strings.isBlank(appName)? APP_DEFAULT_NAME: appName;
+        result.getPlatform().setAppName(appName);
+      }
+      if(!Strings.isBlank(result.getPlatform().getPlatformName())) {
+        // add | to the end of the platformName
+        result.getPlatform().setPlatformName(result.getPlatform().getPlatformName() + " |");
+      }
+    }
```

Comment:
Hi @shenqinb-star @amhudson @gchickma should this just be defaulted in the Settings via the Loader?

I'm inclined to say that we should not be defaulting in the code.

Reply:
@tlawrie, the source code logic is
1. if appName exists in the Settings, it will default to Settings
2. if appName does not exist in the Settings, it will default to "Boomerang Flow" in the code.
Do you think is this logic more strong?

Reply:
I think my concern here is we have been working to remove any hard coded strings from the code that hold a name.

Am I missing a scenario where the default wouldn't be in Settings from the loader?

Reply:
@tlawrie, currently, Flow for CICD doesn't have the appName in the settings.

Reply:
if we default to Settings, we need to have loader to add AppName in settings for Core Flow and CICD Flow.

Reply:
@tlawrie hi tyson, I have changed the code to default appName in settings